test/vault/auth/userpass_test.clj

```diff
@@ -0,0 +1,42 @@
+(ns vault.auth.userpass-test
+  (:require
+    [clojure.test :refer [deftest is testing]]
+    [vault.auth :as auth]
+    [vault.auth.userpass :as userpass]
+    [vault.client :as vault]
+    [vault.integration :refer [with-dev-server cli]]))
+
+
+(defn- assert-authenticated-map
+  [auth]
+  (is (string? (:accessor auth)))
+  (is (string? (:client-token auth)))
+  (is (pos-int? (:lease-duration auth))))
+
+
+(deftest ^:integration http-api
+  (with-dev-server
+    (testing "login"
+      (testing "with default mount"
+        (cli "auth" "enable" "userpass")
+        (cli "write" "auth/userpass/users/foo" "password=bar")
+        (let [original-auth-info (vault/auth-info client)
+              response (userpass/login client "foo" "bar")
+              auth-info (vault/auth-info client)]
+          (assert-authenticated-map response)
+          (is (= (:client-token response)
+                 (::auth/client-token auth-info)))
+          (is (not= (::auth/client-token original-auth-info)
+                    (::auth/client-token auth-info)))))
+      (testing "with alternate mount"
```

Comment:
I did full default/alternate mount tests for kv1 but ultimately decided it was probably not worth the duplication. I wound up going for a simpler test that just exercises `with-mount` and checks for `::userpass/mount` on the client instead. Up to you though, you've already written this.

Reply:
Happy to shorten them so this verbose way of testing isn't cargo-culted

Reply:
Eh I decided to keep it but added assertions on the `::userpass/mount` and removed the check of the original auth info vs the new auth info from the `alternate mount` test at least.